### PR TITLE
 Add Jackson framework for Kotlin

### PR DESCRIPTION
**Pull request by Murnto**

### CODE REVIEW

File: src/quicktype-core/language/Kotlin.ts [hunk 1]
```diff
@@ -63,7 +73,17 @@ export class KotlinTargetLanguage extends TargetLanguage {
...
     ): ConvenienceRenderer {
-        return new KotlinRenderer(this, renderContext, getOptionValues(kotlinOptions, untypedOptionValues));
+        const options = getOptionValues(kotlinOptions, untypedOptionValues);
+
+        switch (options.framework) {
+            case Framework.None:
+                return new KotlinRenderer(this, renderContext, options);
+            case Framework.Klaxon:
+                return new KotlinKlaxonRenderer(this, renderContext, options);
+            default:
```
Comment (schani) [inline src/quicktype-core/language/Kotlin.ts]: `return assertNever(options);` should work here.

File: src/quicktype-core/language/Kotlin.ts [hunk 2]
```diff
@@ -696,3 +699,256 @@ export class KotlinKlaxonRenderer extends KotlinRenderer {
...
+    addSerializer(k.java as Class<T>, object : StdSerializer<T>(k.java as Class<T>) {
+        override fun serialize(value: T, gen: JsonGenerator, provider: SerializerProvider) = gen.writeRawValue(toJson(value))
+    })
+    addDeserializer(k.java as Class<T>, object : StdDeserializer<T>(k.java as Class<T>) {
+        override fun deserialize(p: JsonParser, ctxt: DeserializationContext) = fromJson(p.readValueAsTree())
+    })
+})`);
+    }
+
+    // TODO
```
Comment (schani) [inline src/quicktype-core/language/Kotlin.ts]: What's the TODO here?

**Comment by schani** [comment]
Getting an error on a negative test case:

```
Expected failure on input /app/test/inputs/schema/intersection.1.fail.no-defaults.json
--
  | {
  | "result": "{\"intersection\":{\"foo\":0.0}}",
  | "cwd": "/app/test/runs/schema-kotlin-jackson-7e7d86"
  | }
  | Fixture threw an exception
  | {
  | "error": {
  | "result": "{\"intersection\":{\"foo\":0.0}}",
  | "cwd": "/app/test/runs/schema-kotlin-jackson-7e7d86"
  | },
  | "sample": {
  | "path": "test/inputs/schema/intersection.schema",
  | "additionalRendererOptions": {},
  | "saveOutput": true
  | },
  | "cwd": "/app/test/runs/schema-kotlin-jackson-7e7d86"
  | }
  | [object Object]
  | at /app/test/test.ts:68:17
  | at step (/app/test/test.ts:32:23)
  | at Object.throw (/app/test/test.ts:13:53)
  | at rejected (/app/test/test.ts:5:65)
  | at <anonymous>
  | at process._tickDomainCallback (internal/process/next_tick.js:228:7)
  | npm ERR! Test failed.  See above for more details.
```

**Comment by Murnto** [comment]
I had forgotten to run the schema-kotlin-jackson test fixture locally. It's still running locally for me, but it appears passed the schema fixtures.

**Comment by schani** [comment]
Thank you so much!

**Pull request merged**